iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnTensors.cpp

```diff
@@ -379,17 +461,28 @@ struct ConstOpConversion : public OpRewritePattern<mhlo::ConstOp> {
 void populateHLOToLinalgOnTensorsConversionPatterns(
     MLIRContext *context, OwningRewritePatternList &patterns) {
   mhlo::populateHLOToLinalgConversionPattern(context, &patterns);
-  patterns.insert<TorchIndexSelectOpConversion, ConstOpConversion,
-                  DepthwiseConvOpConversion>(context);
+  patterns.insert<ConstOpConversion, DepthwiseConvOpConversion,
+                  TorchIndexSelectOpConversion>(context);
 }
 
-std::unique_ptr<OperationPass<FuncOp>> createHLOToLinalgOnTensorsPass() {
-  return std::make_unique<ConvertHLOToLinalgOnTensorsPass>();
+static llvm::cl::opt<bool> clUseLinalgOnTensorsPath(
+    "iree-linalg-on-tensors-path",
+    llvm::cl::desc("Convert from MHLO to Linalg on tensors for linalg on "
+                   "tensor codegen path"),
+    llvm::cl::init(false));
+
+std::unique_ptr<OperationPass<FuncOp>> createHLOToLinalgOnTensorsPass(
+    bool useLinalgOnTensorsPath) {
+  return std::make_unique<ConvertHLOToLinalgOnTensorsPass>(
+      useLinalgOnTensorsPath);
 }
 
 static PassRegistration<ConvertHLOToLinalgOnTensorsPass> legalize_pass(
     "iree-codegen-hlo-to-linalg-on-tensors",
-    "Convert from XLA-HLO ops to Linalg ops on tensors");
+    "Convert from XLA-HLO ops to Linalg ops on tensors", []() {
+      return std::make_unique<ConvertHLOToLinalgOnTensorsPass>(
```

Comment:
I think this is gonna cause ASAN issue because of initialization order? (Both this `PassRegistration` and the `clUseLinalgOnTensorsPath` are global static. We had the fun over that in `CodeGenOptionUtils.cpp` previously.

Reply:
Yes, I tried a different thing. Adding a separate flag which enables the Linalg On tensors path by default, but that wasnt recognized by iree-opt. Ill ask on Discord what to do here.